packages/core/src/TxBuilders/TxBuilder.Lucid.V3.class.ts

```diff
@@ -226,6 +188,21 @@ export class TxBuilderLucidV3 extends TxBuilder {
     return this.settingsUtxos;
   }
 
+  /**
+   * Utility function to get the max scooper fee amount, which is defined
+   * in the settings UTXO datum.
+   *
+   * @returns {bigint} The maxScooperFee as defined by the settings UTXO.
+   */
+  public async getMaxScooperFeeAmount(): Promise<bigint> {
+    const [settings] = await this.getAllSettingsUtxos();
+    const { baseFee, simpleFee } = Data.from(
+      settings.datum as string,
+      SettingsDatum
+    );
+    return baseFee + simpleFee;
+  }
```

Comment:
`try` `catch` here?

Reply:
We don't want a try/catch, because if there is an issue with deserializing the datum, it should fail the whole transaction. However, I did add a condition that if the settings utxo isn't present, we fall back to 1 ADA.